pkg/apiserver/etcd_config.go

```diff
@@ -110,6 +111,19 @@ func (c completedEtcdConfig) NewServer() (*ServiceCatalogAPIServer, error) {
 		glog.V(4).Infof("Installing API group %v", provider.GroupName())
 		if err := s.GenericAPIServer.InstallAPIGroup(groupInfo); err != nil {
 			glog.Fatalf("Error installing API group %v: %v", provider.GroupName(), err)
+		} else {
+			// we've sucessfully installed, so hook the stopCh to the destroy func of all the sucessfully installed apigroups
+			for _, mappings := range groupInfo.VersionedResourcesStorageMap { // gv to resource mappings
+				for _, storage := range mappings { // resource name (brokers, brokers/status) to backing storage
+					go func() {
+						s, ok := storage.(*registry.Store)
+						if ok {
+							<-stopCh
+							s.DestroyFunc()
```

Comment:
will this handle subresources correctly as well? They share the store with the main resource and calling DestroyFunc twice leads to panics.

Reply:
Probably not. I did not know that was a restriction (Is this documented?). I've always been very confused about how the sharing of a store happens and how status isn't just overwriting fields in the existing store. It's very confusing.

Is the panic that happens caught and silenced? I don't see it coming up through to kill my tests.

Reply:
Compare how @frobware solved this: https://github.com/kubernetes/kubernetes/pull/50690/files#diff-bec0b0771d0301badbb8737605f24201R81

So there is no proper handling whatsoever. @frobware was fighting with this, until we realized it's due to subresources that share parts with the main resource.